docs/architecture/adr-043-nft-module.md

```diff
@@ -0,0 +1,413 @@
+
+- irismod/nft and modules/incubator/nft
+- CW721
+- DID NFTs
+- interNFT
+
+Since NFTs functions/use cases are tightly connected with their logic, it is almost impossible to support all the NFTs' use cases in one Cosmos SDK module by defining and implementing different transaction types.
+
+Considering generic usage and compatibility of interchain protocols including IBC and Gravity Bridge, it is preferred to have a generic NFT module design which handles the generic NFTs logic.
+
+This design idea can enable composability that application-specific functions should be managed by other modules on Cosmos Hub or on other Zones by importing the NFT module.
+
+The current design is based on the work done by [IRISnet team](https://github.com/irisnet/irismod/tree/master/modules/nft) and an older implementation in the [Cosmos repository](https://github.com/cosmos/modules/tree/master/incubator/nft).
+
+## Decision
+
+We will create a module `x/nft`, which contains the following functionality:
+
+- Store and transfer NFTs utilizing `x/bank`.
+- Mint and burn NFTs.
+- Use `x/authz` to implement ERC721-style authorization.
+- Query NFTs and their supply information.
+
+### Types
+
+#### Metadata
+
+We define a `Metadata` model for `NFT Type`, which is comparable to an ERC721 contract on Ethereum, under which a collection of `NFT`s can be created and managed.
+
+```protobuf
+message Metadata {
+  string type          = 1; // Required, unique key, alphanumeric
+  string name          = 2;
+  string symbol        = 3;
+  string description   = 4;
+  bool mint_restricted = 5;
+  bool edit_restricted = 6;
+}
+```
+
+- The `type` is the identifier of the NFT type/class.
+- The `name` is a descriptive name of this NFT type.
+- The `symbol` is the symbol usually shown on exchanges for this NFT type.
+- The `description` is a detailed description of this NFT type.
+- The `mint_restricted` flag, if set to true, indicates that only the issuer of this type can mint NFTs for it.
+- The `edit_restricted` flag, if set to true, indicates that NFTs of this type cannot be edited once minted.
+
+#### NFT
+
+We define a general model for `NFT` as follows.
+
+```protobuf
+message NFT {
+  string type              = 1; // The type of this NFT
+  string id                = 2; // The identifier of this NFT
+  string uri               = 3; 
+  google.protobuf.Any data = 4;
+}
+```
+
+The NFT conforms to the following specifications:
+
+- The `id` is an immutable field used as a unique identifier within the scope of an NFT type. It is specified by the creator of the NFT and may be expanded to use DID in the future. NFT identifiers don't currently have a naming convention but can be used in association with existing Hub attributes, e.g., defining an NFT's identifier as an immutable Hub address allows its integration into existing Hub account management modules.
```

Comment:
How about this:
```suggestion
- The `id` is an immutable field used as a unique identifier within the scope of an NFT type. It can be automatically managed or specified by the creator of the NFT. `id` can be a DID. NFT identifiers don't currently have a naming convention but can be used in association with existing Hub attributes, e.g., defining an NFT's identifier as an immutable Hub address allows its integration into existing Hub account management modules.
```